config/monitoring/prometheus/monitor.yaml

```diff
@@ -12,7 +12,10 @@ spec:
     - path: /metrics
       port: https
       scheme: https
-      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      authorization:
+        credentials:
+          key: token
+          name: metrics-reader
```

Comment:
The only other reference to `metrics-reader` I found is in this ClusterRole https://github.com/konflux-ci/build-service/blob/273138dd021ce565cb5f1fd7ab674f729d0f93f4/config/rbac/auth_proxy_client_clusterrole.yaml#L2-L4

Does the Secret get created automatically somehow?

Reply:
Ah, the secret I'm seeing isn't made in this repository's configuration.  I was seeing references to it on infra-deployments, so maybe this should live over in that repository?

Alternatively, we could move the manifests of the secret, the service account, and its associated roles and role bindings into this repository.

Reply:
If those manifests are not something specific to the Konflux instance deployed from infra-deployments but make sense in general, let's move them here :+1:

Reply:
@sadlerap would you be able to move them in this PR, or should we leave that to a follow-up?

Reply:
I've got a local branch that has them moved from infra-deployments to here, but I haven't been able to test it yet.  I don't think merging this as-is will break anything, since [the upstream konflux deployments override that service monitor](https://github.com/konflux-ci/konflux-ci/blob/main/konflux-ci/build-service/monitoring/monitoring.coreos.com_v1_servicemonitor_build-service-controller-manager-metrics-monitor.yaml) but I'm not entirely certain if I've accounted for all deployments.

Reply:
Fixed in 37ed5398.